src/plugins/intel_gpu/src/graph/graph_optimizer/mark_state_init_subgraphs.cpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "read_value_inst.h"
+#include "pass_manager.h"
+#include <queue>
+
+#include "intel_gpu/graph/program.hpp"
+
+using namespace cldnn;
+
+void mark_state_init_subgraphs::mark_node(program_node* node) {
+    if (node->is_in_state_init_subgraph())
+        return;
+    if (!node->is_type<read_value>())
+        return;
+
+    const auto& variable_id = node->as<read_value>().get_primitive()->variable_id;
+    node->set_state_variable_id_of_init_subgraph(variable_id);
+
+    std::queue<program_node*> q;
+    q.push(node);
+
+    auto can_be_marked = [&](const program_node* dep_node, const program_node* cur_node) {
+        for (auto& u : dep_node->get_users()) {
+            if (u == cur_node)
```

Comment:
Looks like this `cur_node` arg is redundant as `cur_node` is supposed to always pass get_state_variable_id_of_init_subgraph check

Reply:
removed `cur_node` arg